java/java.editor.lib/src/org/netbeans/editor/ext/java/JavaFoldManager.java

```diff
@@ -43,6 +43,8 @@ public abstract class JavaFoldManager implements FoldManager {
     public static final FoldType JAVADOC_FOLD_TYPE = JavaElementFoldManager.JAVADOC_FOLD_TYPE;
 
     public static final FoldType CODE_BLOCK_FOLD_TYPE = JavaElementFoldManager.CODE_BLOCK_FOLD_TYPE;
```

Comment:
Maybe I did not express the concerns well: The point here is that before the change the user of `JavaFoldManager.CODE_BLOCK_FOLD_TYPE` gets a `FoldType` instance whose `code() == "method"`. That affects e.g. what setting controls the fold type's default expansion state. The user may have `fold.method.defaultCollapsed=true` (just example name) in his settings.

Now with this PR mergd, that unchanged fold producing code that link to `JavaFoldManager.CODE_BLOCK_FOLD_TYPE` will start to receive a `FoldType` with 'code() == "code-block"` ... which means the saved settings won't have suddenly the same effects.

By changing the (potentially) shared definition of `CODE_BLOCK_FOLD_TYPE`, we're breaking settings / semantic compatibility. Yes, it's very frustrating, as a constant named CODE_ was misused for actually method level folds...

Reply:
Not sure what is better -  to preserve the backward semantic compatibility and potentially cause a confusion from non intuitive names or otherwise.

Reply:
Ok, I understand the concerns. Do we have some numbers how many people are still using java.editor.lib? I mean isn't it correct to remove deprecated stuff after years of new releases? Since when is the java.editor.lib deprecated?

Reply:
Is there anything what I can do more? Honestly I don't want to decide, because I don't know how much developers are still using the old java.editor.lib but I mean as I mentioned before, it is deprecated. So I don't see a problem there to change this and to make an help article or smth else to show them the alternative when they want to update. Just my 2 cents.

Reply:
@Chris2011 - even though APIs get deprecated, I don't remember a situation when an API was actually *dropped*, breaking backward compatibility - at least not in such (semantically) 'central' module as java editor. In the past some APIs were broken (see that many modules have /1 or /2 suffixes, which indicates a breaking change), but even during "headless split" (cca NetBeans 8.2) we did not actually drop APIs: see the `*.compat` modules that contain compatibility layer for old applications. Sadly the (deprecated) `java.editor.lib` is such official API with 'forever' commitment.

But I have made a mistake in my reasoning ! If I suppose that an ancient unmantained and useful module still uses the old `editor.lib` APIs, it won't ship that library. So if we leave the (deprecated, misnamed) `JavaFoldManager.CODE_BLOCK_FOLD_TYPE` and relay it to the new, proper-named (but nonpublic) JavaElementFoldManager.METHOD_BLOCK_FOLD_TYPE.  So the ancient client links to the misnamed symbol - but from a 'new version of deprecated java.editor.lib`, thus getting the fold still labeled "method". No semantic change, no settings change. And NB codebase can now use properly named `CODE_BLOCK` and `METHOD_BLOCK`.  Old code lives with past sins, new code uses proper names and even can correctly read the old settings !

Re FoldCreator: the 'createCodeBlockFold` seems to be also now called for initializers and static initializers: we should preserve the fold behaviour here IMHO - that an initializer's fold is controlled by method's settings - but then `createMethodFold` would be awkward for an initializer - wouldn't it ? BTW - isn't there a typo ? The other `FoldCreator` methods use singular nouns but `createMethodsFold` uses plural.

Reply:
So I understand your point of backward compatibilty. For me I just want to know exactly, what I need to do to get this PR merged. As far as I understand you correct, I need to change some code to map some behavior from the generic code_block_fold_type to the more specific method_block_fold_type, didn't I do this?

Reply:
@sdedic hopefully resolved.